core/src/main/scala/com/raphtory/algorithms/temporal/Ancestors.scala

```diff
@@ -58,9 +68,19 @@ class Ancestors(seed: String, time: Long, delta: Long = Long.MaxValue, directed:
                 vertex.setState("ancestor", true)
                 val inEdges    = (if (directed) vertex.getInEdges() else vertex.getEdges())
                   .filter(e => e.earliestActivity().time < latestTime)
-                  .filter(e => e.lastActivityBefore(time).time > latestTime - delta)
+                  .filter(e =>
+                    e.lastActivityBefore(time) match {
+                      case Some(event) => event.time > time - delta
+                      case None        => false
+                    }
+                  )
                 inEdges
-                  .foreach(e => vertex.messageVertex(e.ID(), e.lastActivityBefore(latestTime).time))
+                  .foreach(e =>
+                    e.lastActivityBefore(time) match {
+                      case Some(event) => vertex.messageVertex(e.ID(), event.time)
+                      case None        =>
+                    }
+                  )
```

Comment:
Same as above, this can be just a `foreach` with a single match where we send the message as appropriate with no need to have a `val`...